templates/registry/registry-dpl.yaml

```diff
@@ -205,15 +206,15 @@ spec:
         - name: REGISTRY_REDIS_PASSWORD
           valueFrom:
             secretKeyRef:
-              name: {{ .Values.redis.external.existingSecret }}
-              key: REDIS_PASSWORD
+              name: {{ .Values.redis.external.existingSecret.name }}
+              key: {{ .Values.redis.external.existingSecret.key }}
         {{- end }}
         {{- if .Values.persistence.imageChartStorage.azure.existingSecret }}
         - name: REGISTRY_STORAGE_AZURE_ACCOUNTKEY
           valueFrom:
             secretKeyRef:
-              name: {{ .Values.persistence.imageChartStorage.azure.existingSecret }}
-              key: AZURE_STORAGE_ACCESS_KEY
+              name: {{ .Values.persistence.imageChartStorage.azure.existingSecret.name }}
+              key: {{ .Values.persistence.imageChartStorage.azure.existingSecret.key }}
```

Comment:
I guess this `{{ .Values.persistence.imageChartStorage.azure.existingSecret.key }}` may need to be changed to `{{ .Values.persistence.imageChartStorage.azure.existingSecret.key | default "AZURE_STORAGE_ACCESS_KEY" }}`